### DISCUSSION TITLE
MPI - Fatal Error

### DISCUSSION THREAD

**Discussion by 12arya129** [idaholab/moose#16799] [Q&A General] [answered]
Hello,

Can anyone help me with this one? I tried using MPI on one of the examples from Moose Virtual Workshop - Summer 2020, it's working fine with no MPI.

Am also attaching core numbers.

Thank you

![Screenshot from 2021-01-23 10-22-53](https://user-images.githubusercontent.com/77883591/105578490-9aeb5380-5da6-11eb-9994-6c1dfa775737.png)
![Screenshot from 2021-01-23 10-21-19](https://user-images.githubusercontent.com/77883591/105578491-9c1c8080-5da6-11eb-836d-3a611bed63b3.png)

**Comment by GiudGiud** [comment, marked as answer]
Hi 

Have you tried the solutions proposed in [stack_overflow_mpich](https://stackoverflow.com/questions/23112515/mpich2-gethostbyname-failed) ?
It has to do with the local hosts.

Best,
Guillaume

**Reply by 12arya129** [reply]
Thanks a lot for the link, yes it worked. I changed that hostname of terminal from "sudo hostnamectl set-hostname xyz" and edited hostname file with "sudo nano /etc/hosts"

**Reply by 12arya129** [reply]
The problem still persists, that earlier code ran but next one gave me the same error. I check the hostname which was same for terminal and the file.
![Screenshot from 2021-01-25 19-37-50](https://user-images.githubusercontent.com/77883591/105716797-56e28500-5f45-11eb-9923-ddfbd12da310.png)

Any suggestions?

**Reply by milljm** [reply]
What is the contents of your hosts file?

```
cat /etc/hosts
```

Also, would it be possible to not post pictures of text? The reason for this, is because when searching for relevant issues on the discussion board, it's impossible to search for text _inside_ pictures, and so your issue never comes up (gethostbyname is no-where in this discussion until just now because I wrote it).

**Reply by 12arya129** [reply]
Oh thanks for the advice. Below are the contents 

127.0.0.1 Arya

# The following lines are desirable for IPv6 capable hosts
::1     ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters

**Reply by fdkong** [reply]
It is interesting. What is the result of  command-line: `hostname`? 

You might then = add whatever result you get from the command line to /etc/hosts
